src/ios/PurchasesPlugin.swift

```diff
@@ -1,10 +1,450 @@
 //
 //  PurchasesPlugin.swift
-//  HelloCordova
+//  PurchasesPlugin
 //
-//  Created by Andrés Boedo on 9/23/20.
+//  Created by Joshua Liebowitz on 6/27/22
 //
-// This file is left intentionally blank,
 // the code for the plugin lives in CDVPurchasesPlugin
 
 import Foundation
+import PurchasesHybridCommon
+
+public class CDVPurchasesPlugin : CDVPlugin {
+
+    private var updatedPurchaserInfoCallbackID: String?
+    private var shouldPurchasePromoProductCallbackID: String?
+    private var defermentBlocks: [RCDeferredPromotionalPurchaseBlock] = []
+
+    @objc(setupPurchases:)
+    func setupPurchases(command: CDVInvokedUrlCommand) {
+        guard let apiKey = command.arguments[0] as? String else {
+            self.sendBadParameterFor(command: command, parameterNamed: "apiKey", expectedType: String.self)
+            return
+        }
+        let appUserID = command.arguments[1] as? String
+        let observerMode = command.arguments[2] as? Bool ?? false
+        let userDefaultsSuiteName = command.arguments[3] as? String
```

Comment:
Do we want to throw an error if these are the wrong type? Or log they are the wrong type? 🤔

Reply:
Maybe? I'll think about it. Logging that as a task now.